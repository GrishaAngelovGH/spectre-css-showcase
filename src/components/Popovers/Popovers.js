const Card = () => (
  <div className='card'>
    <div className='card-header'>
      <div className='card-title h5'>Apple</div>
      <div className='card-subtitle text-gray'>Software and hardware</div>
    </div>
    <div className='card-body'>To make a contribution to the world by making tools for the mind that advance humankind.</div>
    <div className='card-footer'>
      <button className='btn btn-primary'>Buy</button>
    </div>
  </div>
)

const Popovers = () => {
  const popovers = ['', 'popover-right', 'popover-bottom', 'popover-left']

  return (
    <div className='columns' style={{ padding: 100 }}>
      {
        popovers.map((v, i) => (
          <div key={i} className='column col-3 col-sm-6 col-mx-auto'>
            <div className={`popover ${v}`}><a className='btn btn-primary' href='#popovers'>{v.length ? v : 'top popover'}</a>
              <div className='popover-container'>
                <Card />
              </div>
            </div>
          </div>
        ))
      }
    </div>
  )
}

export default Popovers