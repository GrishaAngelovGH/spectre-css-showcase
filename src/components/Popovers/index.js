import Popovers from './Popovers'

export default Popovers