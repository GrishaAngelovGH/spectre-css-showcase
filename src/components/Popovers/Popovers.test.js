import { render } from '@testing-library/react'

import Popovers from './Popovers'

test('should render Popovers component', () => {
  const view = render(<Popovers />)

  expect(view).toMatchSnapshot()
})