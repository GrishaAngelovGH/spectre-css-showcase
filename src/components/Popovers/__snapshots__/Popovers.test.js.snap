// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Popovers component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="columns"
        style="padding: 100px;"
      >
        <div
          class="column col-3 col-sm-6 col-mx-auto"
        >
          <div
            class="popover "
          >
            <a
              class="btn btn-primary"
              href="#popovers"
            >
              top popover
            </a>
            <div
              class="popover-container"
            >
              <div
                class="card"
              >
                <div
                  class="card-header"
                >
                  <div
                    class="card-title h5"
                  >
                    Apple
                  </div>
                  <div
                    class="card-subtitle text-gray"
                  >
                    Software and hardware
                  </div>
                </div>
                <div
                  class="card-body"
                >
                  To make a contribution to the world by making tools for the mind that advance humankind.
                </div>
                <div
                  class="card-footer"
                >
                  <button
                    class="btn btn-primary"
                  >
                    Buy
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="column col-3 col-sm-6 col-mx-auto"
        >
          <div
            class="popover popover-right"
          >
            <a
              class="btn btn-primary"
              href="#popovers"
            >
              popover-right
            </a>
            <div
              class="popover-container"
            >
              <div
                class="card"
              >
                <div
                  class="card-header"
                >
                  <div
                    class="card-title h5"
                  >
                    Apple
                  </div>
                  <div
                    class="card-subtitle text-gray"
                  >
                    Software and hardware
                  </div>
                </div>
                <div
                  class="card-body"
                >
                  To make a contribution to the world by making tools for the mind that advance humankind.
                </div>
                <div
                  class="card-footer"
                >
                  <button
                    class="btn btn-primary"
                  >
                    Buy
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="column col-3 col-sm-6 col-mx-auto"
        >
          <div
            class="popover popover-bottom"
          >
            <a
              class="btn btn-primary"
              href="#popovers"
            >
              popover-bottom
            </a>
            <div
              class="popover-container"
            >
              <div
                class="card"
              >
                <div
                  class="card-header"
                >
                  <div
                    class="card-title h5"
                  >
                    Apple
                  </div>
                  <div
                    class="card-subtitle text-gray"
                  >
                    Software and hardware
                  </div>
                </div>
                <div
                  class="card-body"
                >
                  To make a contribution to the world by making tools for the mind that advance humankind.
                </div>
                <div
                  class="card-footer"
                >
                  <button
                    class="btn btn-primary"
                  >
                    Buy
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="column col-3 col-sm-6 col-mx-auto"
        >
          <div
            class="popover popover-left"
          >
            <a
              class="btn btn-primary"
              href="#popovers"
            >
              popover-left
            </a>
            <div
              class="popover-container"
            >
              <div
                class="card"
              >
                <div
                  class="card-header"
                >
                  <div
                    class="card-title h5"
                  >
                    Apple
                  </div>
                  <div
                    class="card-subtitle text-gray"
                  >
                    Software and hardware
                  </div>
                </div>
                <div
                  class="card-body"
                >
                  To make a contribution to the world by making tools for the mind that advance humankind.
                </div>
                <div
                  class="card-footer"
                >
                  <button
                    class="btn btn-primary"
                  >
                    Buy
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="columns"
      style="padding: 100px;"
    >
      <div
        class="column col-3 col-sm-6 col-mx-auto"
      >
        <div
          class="popover "
        >
          <a
            class="btn btn-primary"
            href="#popovers"
          >
            top popover
          </a>
          <div
            class="popover-container"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="card-title h5"
                >
                  Apple
                </div>
                <div
                  class="card-subtitle text-gray"
                >
                  Software and hardware
                </div>
              </div>
              <div
                class="card-body"
              >
                To make a contribution to the world by making tools for the mind that advance humankind.
              </div>
              <div
                class="card-footer"
              >
                <button
                  class="btn btn-primary"
                >
                  Buy
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="column col-3 col-sm-6 col-mx-auto"
      >
        <div
          class="popover popover-right"
        >
          <a
            class="btn btn-primary"
            href="#popovers"
          >
            popover-right
          </a>
          <div
            class="popover-container"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="card-title h5"
                >
                  Apple
                </div>
                <div
                  class="card-subtitle text-gray"
                >
                  Software and hardware
                </div>
              </div>
              <div
                class="card-body"
              >
                To make a contribution to the world by making tools for the mind that advance humankind.
              </div>
              <div
                class="card-footer"
              >
                <button
                  class="btn btn-primary"
                >
                  Buy
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="column col-3 col-sm-6 col-mx-auto"
      >
        <div
          class="popover popover-bottom"
        >
          <a
            class="btn btn-primary"
            href="#popovers"
          >
            popover-bottom
          </a>
          <div
            class="popover-container"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="card-title h5"
                >
                  Apple
                </div>
                <div
                  class="card-subtitle text-gray"
                >
                  Software and hardware
                </div>
              </div>
              <div
                class="card-body"
              >
                To make a contribution to the world by making tools for the mind that advance humankind.
              </div>
              <div
                class="card-footer"
              >
                <button
                  class="btn btn-primary"
                >
                  Buy
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="column col-3 col-sm-6 col-mx-auto"
      >
        <div
          class="popover popover-left"
        >
          <a
            class="btn btn-primary"
            href="#popovers"
          >
            popover-left
          </a>
          <div
            class="popover-container"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="card-title h5"
                >
                  Apple
                </div>
                <div
                  class="card-subtitle text-gray"
                >
                  Software and hardware
                </div>
              </div>
              <div
                class="card-body"
              >
                To make a contribution to the world by making tools for the mind that advance humankind.
              </div>
              <div
                class="card-footer"
              >
                <button
                  class="btn btn-primary"
                >
                  Buy
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
